packages/react-native/React/Views/UIView+React.m

```diff
@@ -203,6 +203,77 @@ - (void)reactSetFrame:(CGRect)frame
 
   self.center = position;
   self.bounds = bounds;
+
+  [self updateTransform];
+}
+
+#pragma mark - Transforms
+
+- (CATransform3D)reactTransform
+{
+  id obj = objc_getAssociatedObject(self, _cmd);
+  return obj != nil ? [obj CATransform3DValue] : CATransform3DIdentity;
+}
+
+- (void)setReactTransform:(CATransform3D)reactTransform
+{
+  objc_setAssociatedObject(
+      self, @selector(reactTransform), @(reactTransform), OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+  [self updateTransform];
+}
+
+- (RCTTransformOrigin)reactTransformOrigin
+{
+  RCTTransformOrigin transformOrigin = {
+    (YGValue){50, YGUnitPercent},
+    (YGValue){50, YGUnitPercent},
+    0
+  };
+  id obj = objc_getAssociatedObject(self, _cmd);
+  if (obj != nil) {
+    [obj getValue:&transformOrigin];
+  }
+  return transformOrigin;
+}
+
+- (void)setReactTransformOrigin:(RCTTransformOrigin)reactTransformOrigin
+{
+  id obj = [NSValue value:&reactTransformOrigin withObjCType:@encode(RCTTransformOrigin)];
+  objc_setAssociatedObject(
+      self, @selector(reactTransformOrigin), obj, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+  [self updateTransform];
+}
+
+- (void)updateTransform
```

Comment:
This can be a static helper method that takes in the UIView, to avoid adding unnecessary category methods.

Reply:
Updated